### DISCUSSION TITLE
Is Snowflake connection column lineage still not supported?

### DISCUSSION THREAD

**Discussion by AryamanMishra** [OpenLineage/OpenLineage#2271] [General]
According to [this](https://openlineage.io/blog/openlineage-snowflake/) article, column lineage on snowflake connected datasets are still under development. Is this still a case or their is a development on it.

**Comment by mobuchowski** [comment]
@AryamanMishra it describes one potential approach that unfortunately did not get too much love. If you use `SnowflakeOperator` in Airflow or connect to Snowflake from Spark, it works.

**Reply by AryamanMishra** [reply]
Hey @mobuchowski thanks for the answer. Currently I was using SqlExecuteQuery Operator and their was no column lineage. And yes as I am working with airflow I will now use SnowflakeOperator itself. Thanks again.
Just one thing more. As SnowflakeOperator class shows depreciated so in future for seeing column lineage we have to use spark connection. That would be the only way then

**Reply by mobuchowski** [reply]
@AryamanMishra could you post the SQL (can be anonymized) that does not have that column-level lineage? Also, is the Airflow and OpenLineage provider version the newest?

**Reply by AryamanMishra** [reply]
Yes I will provide some context about my experiment too.
I was working on openlineage airflow integration and I used [this](https://pypi.org/project/openlineage-airflow/) module to set it up.
So just pip installed the module and set some env variables and it was up and running.
Then I was working on PostgresOperator and just wrote an airflow dag to join 2 tables. The dag contains a task with sql query as - 
```
CREATE TABLE IF NOT EXISTS employees_department_1  AS
SELECT e.employee_id,e.first_name, e.last_name, e.salary, d.department_name
FROM employees e
JOIN departments d ON e.department_id = d.department_id
WHERE e.department_id = 1

consider employees and department tables already exists.
```
when I looked into metadata in marquez UI i can see the new table and its column lineage correctly like this.
![Screenshot from 2023-11-21 23-05-17](https://github.com/OpenLineage/OpenLineage/assets/61847192/8dc5afb1-a1a2-48ba-b4d1-dedeb2bfbe9c)

When I tried the same thing on snowflake datasets it wasnt showing anything.
```
CREATE TABLE IF NOT EXISTS AIRFLOW_TEST.PUBLIC.CUST_CUSTADDRESS_JOINED_TABLE AS
SELECT *
FROM AIRFLOW_TEST.PUBLIC.CUSTOMER_100 AS C
JOIN AIRFLOW_TEST.PUBLIC.CUSTOMER_ADDRESS_100 AS CA ON C.C_CURRENT_ADDR_SK = CA.CA_ADDRESS_SK
```
Here I am doing the same join operation on sample public snowflake tables. Consider that the tables CUSTOMER_100 and CUSTOMER_ADDRESS_100 are created beforehand from sample tables in snowflake public datasets. But when I view this newly created table on marquez it shows no column lineage.
Also as you said to use SnowflakeOperator I did that too but the result is same.
Am I missing something. Feel free to ask more if I was not adequate with my explaination.
Thanks again

**Reply by AryamanMishra** [reply]
Hey i just found the reason why this is happening. whenever i do SELECT * operation for making new table or running a join task, it never identifies column lineage. so the workaround is to specify columns after SELECT statement. But this wouldnt be feasible for too many columns

**Reply by mobuchowski** [reply]
Yeah - it would require _the integration_ to query the database to get an idea of what actual columns get requested. For now we've decided that we'd prefer to not do that.
